File: frontend/proyecto-adopta-frontend/src/services/citaService.js
import api from './api'

export const citaService = {

  async listar() {
    return api.get('/api/citas')
  },

  async programar(idSolicitud, fechaCita, observaciones) {
    return api.post('/api/citas', { idSolicitud, fechaCita, observaciones })
  },

  async borrar(idCita) {
    return api.delete(`/api/citas/${idCita}`)
  }
}

File: backend/AdoptaUnCompaneroApplication-3/src/main/java/com/daw/adoptauncompanero/servicio/interfaces/CitaService.java
package com.daw.adoptauncompanero.servicio.interfaces;

import java.time.LocalDateTime;
import java.util.List;

import com.daw.adoptauncompanero.dtos.CitaDTO;

public interface CitaService {

    Integer programarCita(Integer idSolicitud, LocalDateTime fechaCita, String observaciones);

    Integer actualizarCita(Integer idCita, LocalDateTime fechaCita, String observaciones);

    Integer borrarCita(Integer idCita);

    List<CitaDTO> listarTodasLasCitas();

    List<CitaDTO> listarCitasPorUsuario(Integer idUsuario);
}